packages/react-instantsearch/src/core/utils.js

```diff
@@ -58,3 +58,6 @@ export function assertFacetDefined(searchParameters, searchResults, facet) {
 export function getDisplayName(Component) {
   return Component.displayName || Component.name || 'UnknownComponent';
 }
+
+const resolved = typeof Promise !== 'undefined' && Promise.resolve();
+export const defer = resolved ? f => { resolved.then(f); } : setTimeout;
```

Comment:
Ok :) I was just curious. It's a very hacky way to use promises :) Especially since we are going to reuse the same promise mutliple times :)